GTest CMake broken under CMake 2.8

extras/gtest/CMakeLists.txt fails with 

```
CMake Error at rapidcheck/extras/gtest/CMakeLists.txt:2 (add_library):     
  Cannot find source file:   
         INTERFACE
  Tried extensions .c .C .c++ .cc .cpp .cxx .m .M .mm .h .hh .h++ .hm .hpp .hxx .in .txx     
```
This is a caused by the using the "INTERFACE" feature which is not available in CMake 2.8.12.  However, the CMakeLists.txt in the root declares a minimum version of CMake 2.8.12.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.8.12)
+cmake_minimum_required(VERSION 3.0)
 project(rapidcheck)
 
 # Don't warn about symbol visibility for static libraries with CMake 3.3 and later.

